config/confighttp/confighttp.go

```diff
@@ -58,6 +58,9 @@ type HTTPClientSettings struct {
 
 	// Auth configuration for outgoing HTTP calls.
 	Auth *configauth.Authentication `mapstructure:"auth,omitempty"`
+
+	// The compression key for supported compression types within collector.
+	Compression middleware.CompressionType `mapstructure:"compression"`
```

Comment:
To fail fast (during config unmarshaling) make CompressionType to implement this interface https://pkg.go.dev/encoding#TextUnmarshaler

Reply:
Thank you for the reply. Fixed it.